src/services/expressions.ts

```diff
@@ -21,37 +20,30 @@ import { getFieldsVariable, getLabelsVariable } from './variableGetters';
  * @param excludeEmpty - if true, the query will exclude empty values for the given streamSelectorName
  */
 export function getTimeSeriesExpr(sceneRef: SceneObject, streamSelectorName: string, excludeEmpty = true): string {
-  const labelsVariable = getLabelsVariable(sceneRef);
   const fieldsVariable = getFieldsVariable(sceneRef);
 
-  let labelExpressionToAdd;
   let metadataExpressionToAdd = '';
   if (excludeEmpty) {
     // `LEVEL_VARIABLE_VALUE` is a special case where we don't want to add this to the stream selector
-    if (streamSelectorName !== LEVEL_VARIABLE_VALUE) {
-      labelExpressionToAdd = { key: streamSelectorName, operator: '!=', value: '' };
-    } else {
+    if (streamSelectorName === LEVEL_VARIABLE_VALUE) {
       metadataExpressionToAdd = `| ${LEVEL_VARIABLE_VALUE} != ""`;
     }
   }
 
-  const labelFilters = [...labelsVariable.state.filters, labelExpressionToAdd].filter(isDefined);
-  const streamSelectors = renderLogQLLabelFilters(labelFilters);
```

Comment:
As far as I can see, renderLogQLLabelFilters() does a lot more than just interpolating label filters. Are we sure about this change?

Reply:
It's only required if we're manually building strings instead of interpolating from the VAR_LABELS variable, which already calls renderLogQLLabelFilters automatically when building expressions, see the `expressionBuilder` method on `AdHocFiltersVariable` class

Reply:
Wouldn't hurt to do some more testing to make sure queries are being built correctly though, and if we find any bugs we should add those as test cases